internal/gcsx/temp_file.go

```diff
@@ -227,8 +226,14 @@ func (tf *tempFile) CheckInvariants() {
 
 func (tf *tempFile) Destroy() {
 	tf.state = fileDestroyed
-	// Throw away the file.
+	// Throw away the file (for anonymous files).
 	tf.f.Close()
+
+	// TODO ezl: move this logic out once we refactor cache disk file handling outside of gcsx.Tempfile
+	// Throw away the cache file and metadata
+	os.Remove(fmt.Sprintf("%s.json", tf.f.Name()))
+	os.Remove(tf.f.Name())
```

Comment:
Same as the comment above. Why do we need to delete this file?

Reply:
Same as above.  We clear these files whenever destroy is called.  These should be moved once we decouple cache files from gcsx.TempFile.